src/sygnet/sygnet_requirements.py

```diff
@@ -13,6 +13,7 @@
 import pandas as pd
 from datetime import datetime
 from tqdm import trange
+import pickle
```

Comment:
Can you also update the [pyproject.toml](https://github.com/tsrobinson/SyGNet/blob/main/pyproject.toml) to add **pickle** as an optional dependency? Perhaps create a new list called 'save' (like we have for vis currently on line 44 of the .toml file).